### PR TITLE
OTG Test - TE-3.2: Traffic Balancing According to Weights

### LINKED ISSUE #569: OTG: TE-3.2: Traffic Balancing According to Weights
- [README](https://github.com/openconfig/featureprofiles/blob/main/feature/gribi/otg_tests/weighted_balancing_test/README.md)
- Init PR #537

### PR DESCRIPTION
**Pull request by octpetre**
OTG test conversion for **_TE-3.2: Traffic Balancing According to Weights_**
Test log output found [here](https://drive.google.com/file/d/1LUkr9LRB61IlqdgSY71RTjbMjKg1hqRM/view?usp=sharing)
Fixes #569

### CODE REVIEW

File: feature/gribi/otg_tests/weighted_balancing_test/port_flap_rebalanced_test.go [hunk 1]
```diff
@@ -152,18 +153,17 @@ func TestPortFlap(t *testing.T) {
 
 		t.Run(testName, func(t *testing.T) {
 			if i < len(atePorts) {
-				t.Logf("Bringing down ate port: %v", atePorts[i])
-				ate.Actions().
-					NewSetPortState().
-					WithPort(atePorts[i]).
-					WithEnabled(false).
-					Send(t)
+				// t.Logf("Bringing down ate port: %v", atePorts[i])
```
Comment (marcushines) [inline feature/gribi/otg_tests/weighted_balancing_test/port_flap_rebalanced_test.go:156]: remove line

File: feature/gribi/otg_tests/weighted_balancing_test/setup_test.go [hunk 2]
```diff
@@ -124,6 +136,7 @@ var (
 		"ate:port9": "192.0.2.34",
 	}
 )
+var ateMac = make(map[string]string)
```
Comment (marcushines) [inline feature/gribi/otg_tests/weighted_balancing_test/setup_test.go:139]: add comment on what this for

File: feature/gribi/otg_tests/weighted_balancing_test/setup_test.go [hunk 3]
```diff
@@ -194,28 +207,36 @@ func configureDUT(t testing.TB, dut *ondatra.DUTDevice) {
 	}
 }
 
+// setDutInterfaceState sets the admin state on the dut interface
```
Comment (marcushines) [inline feature/gribi/otg_tests/weighted_balancing_test/setup_test.go:210]: DUT should be capitalized

File: feature/gribi/otg_tests/weighted_balancing_test/setup_test.go [hunk 4]
```diff
@@ -194,28 +207,36 @@ func configureDUT(t testing.TB, dut *ondatra.DUTDevice) {
...
+func setDutInterfaceState(t testing.TB, dut *ondatra.DUTDevice, p *ondatra.Port, state bool) {
```
Comment (marcushines) [inline feature/gribi/otg_tests/weighted_balancing_test/setup_test.go:211]: this should be t.Helper()

File: feature/gribi/otg_tests/weighted_balancing_test/setup_test.go [hunk 5]
```diff
@@ -194,28 +207,36 @@ func configureDUT(t testing.TB, dut *ondatra.DUTDevice) {
...
+	i := &telemetry.Interface{Name: ygot.String(p.Name())}
+	i.Enabled = ygot.Bool(state)
+	dc.Interface(p.Name()).Update(t, i)
+}
+
 // configureATE configures the topology of the ATE.
-func configureATE(t testing.TB, ate *ondatra.ATEDevice) *ondatra.ATETopology {
-	top := ate.Topology().New()
-	for _, ap := range ate.Ports() {
+func configureATE(t testing.TB, ate *ondatra.ATEDevice) gosnappi.Config {
```
Comment (marcushines) [inline feature/gribi/otg_tests/weighted_balancing_test/setup_test.go:219]: t.Helper()

File: feature/gribi/otg_tests/weighted_balancing_test/setup_test.go [hunk 6]
```diff
@@ -390,3 +431,29 @@ func debugGRIBI(t testing.TB, dut *ondatra.DUTDevice) {
...
+	buf := new(bytes.Buffer)
+	err = binary.Write(buf, binary.BigEndian, convMac)
+	if err != nil {
+		return "", err
+	}
+	newMac := net.HardwareAddr(buf.Bytes()[2:8])
+	return newMac.String(), nil
+}
+
+// waitOtgArpEntry ensures that ARP entries are present on the tx otg interface and traffic could be started
```
Comment (marcushines) [inline feature/gribi/otg_tests/weighted_balancing_test/setup_test.go:452]: OTG in comment

File: feature/gribi/otg_tests/weighted_balancing_test/setup_test.go [hunk 7]
```diff
@@ -390,3 +431,29 @@ func debugGRIBI(t testing.TB, dut *ondatra.DUTDevice) {
...
+func waitOTGARPEntry(t *testing.T) {
```
Comment (marcushines) [inline feature/gribi/otg_tests/weighted_balancing_test/setup_test.go:453]: t.Helper()

File: feature/gribi/otg_tests/weighted_balancing_test/setup_test.go [hunk 8]
```diff
@@ -390,3 +431,29 @@ func debugGRIBI(t testing.TB, dut *ondatra.DUTDevice) {
...
+	ate := ondatra.ATE(t, "ate")
+	ate.OTG().Telemetry().Interface(ateSrcPort+".Eth").Ipv4NeighborAny().LinkLayerAddress().Watch(
+		t, time.Minute, func(val *otgtelemetry.QualifiedString) bool {
```
Comment (marcushines) [inline feature/gribi/otg_tests/weighted_balancing_test/setup_test.go:456]: likely time.Minute should be configurable? or?

File: feature/gribi/otg_tests/weighted_balancing_test/setup_test.go [hunk 9]
```diff
@@ -390,3 +431,29 @@ func debugGRIBI(t testing.TB, dut *ondatra.DUTDevice) {
 		t.Log("afts value not present")
 	}
 }
+
+// incrementedMac uses an mac string and increments it by the given i
```
Comment (marcushines) [inline feature/gribi/otg_tests/weighted_balancing_test/setup_test.go:435]: incrementMAC

Comment (marcushines) [inline feature/gribi/otg_tests/weighted_balancing_test/setup_test.go:435]: // incrementMAC increments the MAC by i.  Returns error if the mac cannot be parsed or overflows the mac address space

Comment (octpetre) [reply feature/gribi/otg_tests/weighted_balancing_test/setup_test.go:435]: Done

Comment (octpetre) [reply feature/gribi/otg_tests/weighted_balancing_test/setup_test.go:435]: Done

File: feature/gribi/otg_tests/weighted_balancing_test/setup_test.go (hunk 4 above)
Comment (octpetre) [reply feature/gribi/otg_tests/weighted_balancing_test/setup_test.go:211]: Done

File: feature/gribi/otg_tests/weighted_balancing_test/setup_test.go (hunk 3 above)
Comment (octpetre) [reply feature/gribi/otg_tests/weighted_balancing_test/setup_test.go:210]: Done

File: feature/gribi/otg_tests/weighted_balancing_test/setup_test.go (hunk 7 above)
Comment (octpetre) [reply feature/gribi/otg_tests/weighted_balancing_test/setup_test.go:453]: Done

File: feature/gribi/otg_tests/weighted_balancing_test/setup_test.go (hunk 5 above)
Comment (octpetre) [reply feature/gribi/otg_tests/weighted_balancing_test/setup_test.go:219]: Added

File: feature/gribi/otg_tests/weighted_balancing_test/setup_test.go (hunk 2 above)
Comment (octpetre) [reply feature/gribi/otg_tests/weighted_balancing_test/setup_test.go:139]: Removed

File: feature/gribi/otg_tests/weighted_balancing_test/setup_test.go (hunk 8 above)
Comment (octpetre) [reply feature/gribi/otg_tests/weighted_balancing_test/setup_test.go:456]: Added the input "timeout"

**Review by sezhang2 — APPROVED** [review]

**Comment by octpetre** [comment]
@sezhang2 could you please merge?

**Pull request merged**